src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,153 @@
+
+    @Test
+    void register_User_Ok() {
+        registrationService.register(user);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void ageIs0_notOk() {
+        user.setAge(0);
+        assertThrows(UserRegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(UserRegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void ageIsNegative_notOk() {
+        user.setAge(-10);
+        assertThrows(UserRegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(UserRegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void shortLogin_notOk() {
+        user.setLogin("login");
+        assertThrows(UserRegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void loginIsEmpty_notOk() {
+        user.setLogin("");
+        assertThrows(UserRegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(UserRegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void shortPassword_notOk() {
+        user.setPassword("word");
+        assertThrows(UserRegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void passwordIsEmpty_notOk() {
+        user.setPassword("");
+        assertThrows(UserRegistrationException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void getAge_Ok() {
+        int expected = 21;
+        int actual = registrationService.register(user).getAge();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void setAge_Ok() {
+        user.setAge(44);
+        int expected = 44;
+        int actual = registrationService.register(user).getAge();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void getLogin_Ok() {
+        String expected = "myLogin";
+        String actual = registrationService.register(user).getLogin();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void setLogin_Ok() {
+        user.setLogin("myName");
+        String expected = "myName";
+        String actual = registrationService.register(user).getLogin();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void getPassword_Ok() {
+        String expected = "myPassword";
+        String actual = registrationService.register(user).getPassword();
+        assertEquals(expected, actual);
+    }
+
+    @Test
```

Comment:
Test methods should focus on testing the functionality of the related class, not the User model class. Consider removing this test and other similar ones that focus on testing the getter and setter methods of the User class.